server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -2907,6 +2909,79 @@ public void testRetrieveUsedSegmentsAndCreatedDatesFetchesEternityForAnyInterval
     Assert.assertEquals(3, resultForEternity.size());
   }
 
+  @Test
+  public void testTimelineWithTombstoneCorePartitionsAs0() throws IOException
+  {
+    final Interval interval = Intervals.of("2020/2021");
+    // Create and commit a tombstone segment
+    final DataSegment tombstoneSegment = createSegment(
+        interval,
+        "version",
+        new TombstoneShardSpec()
+    );
+
+    final Set<DataSegment> tombstones = new HashSet<>(Collections.singleton(tombstoneSegment));
+    Assert.assertTrue(coordinator.commitSegments(tombstones).containsAll(tombstones));
+
+    // Create and commit a data segment
+    final DataSegment dataSegment = createSegment(
```

Comment:
I think it may be sufficient to have just one test which allocates and commits a segment using the methods in IndexerSqlMetadataStorageCoordinator. The test would succeed with the changes in the PR and fail without them

Reply:
Thanks. Added allocate pending segment and commit for the data segment. Kept it as two separate tests still since the old generation tombstones (with 1 core partition) have been around for awhile, so the test behavior shows what to expect.